client/termsetting/handlers/samplelst.js

```diff
@@ -118,9 +122,151 @@ export function getSamplelstTW(groups, name = 'groups', groupsetting = {}) {
+	const parentMenu = new Menu({ padding: '5px' })
+
+	const menuDiv = parentMenu.d.append('div')
+
+	let row = menuDiv.append('div')
+
+	addMatrixMenuItems(parentMenu, menuDiv, samplelstTW, app, state)
+	if (allowedTermTypes.includes('survival')) {
+		const survivalDiv = menuDiv
+			.append('div')
+			.attr('class', 'sja_menuoption sja_sharp_border')
+			.html('Compare survival&nbsp;&nbsp;›')
+			.on('click', e => {
+				const state = {
+					nav: { header_mode: 'hide_search' },
+					tree: { usecase: { target: 'survival', detail: 'term' } }
+				}
+				showTermsTree(
+					survivalDiv,
+					term => {
+						openSurvivalPlot(term, samplelstTW, app)
+					},
+					app,
+					parentMenu,
+					state
+				)
+			})
+	}
+	const summarizeDiv = menuDiv
+		.append('div')
+		.attr('class', 'sja_menuoption sja_sharp_border')
+		.html('Summarize')
+	summarizeDiv
+		.insert('div')
+		.html('›')
+		.style('float', 'right')
+
+	summarizeDiv.on('click', async e => {
+		showTermsTree(
+			summarizeDiv,
+			term => {
+				openSummaryPlot(term, samplelstTW, app)
+			},
+			app,
+			parentMenu
+		)
+	})
+	row = menuDiv
+		.append('div')
+		.attr('class', 'sja_menuoption sja_sharp_border')
+		.text('Delete variable')
+		.on('click', event => {
+			deleteCallback()
+			parentMenu.hide()
+		})
+
+	parentMenu.show(event.clientX, event.clientY)
+}
+
+export async function openSurvivalPlot(term, term2, app, id) {
+	let config = {
+		chartType: 'survival',
+		term,
+		term2
+	}
+	if (id) config.insertBefore = id
+	await app.dispatch({
+		type: 'plot_create',
+		config
+	})
+}
+
+export async function openSummaryPlot(term, samplelstTW, app, id) {
+	// barchart config.term{} name is confusing, as it is actually a termsetting object, not term
+	// thus convert the given term into a termwrapper
+	// tw.q can be missing and will be filled in with default setting
+	const tw = { id: term.id, term }
+	let config = {
+		chartType: 'summary',
+		childType: 'barchart',
+		term: tw,
+		term2: samplelstTW
+	}
+	if (id) config.insertBefore = id
+	await app.dispatch({
+		type: 'plot_create',
+		config
+	})
+}
+
+export async function showTermsTree(div, callback, app, parentMenu, state = { tree: { usecase: { detail: 'term' } } }) {
+	const menu = new Menu({ padding: '5px', offsetX: 170, offsetY: -34 })
+	menu.showunderoffset(div.node())
+	const termdb = await import('../../termdb/app')
+	termdb.appInit({
+		holder: menu.d,
+		vocabApi: app.vocabApi,
+		state,
+		tree: {
+			click_term: term => {
+				callback(term)
+				menu.hide()
+				parentMenu.hide()
+			}
+		}
+	})
+}
+
+export function addMatrixMenuItems(menu, menuDiv, tw, app, state) {
+	if (state.matrixplots) {
+		for (const plot of state.matrixplots) {
+			menuDiv
+				.append('div')
+				.attr('class', 'sja_menuoption sja_sharp_border')
+				.text(plot.name)
+				.on('click', async () => {
+					const config = await dofetch3('termdb', {
+						body: {
+							for: 'matrix',
+							getPlotDataByName: plot.name,
```

Comment:
i realized this is copied over from matrix.js. in the next branch can replace both placed with a new api method